Source/drlg_l1.cpp

```diff
@@ -1352,19 +1352,17 @@ static BOOL L5checkRoom(int x, int y, int width, int height)
 
 static void L5roomGen(int x, int y, int w, int h, int dir)
 {
-	int num, dirProb;
 	BOOL ran, ran2;
 	int width, height, rx, ry, ry2;
 	int cw, ch, cx1, cy1, cx2;
 
-	dirProb = random_(0, 4);
+	int dirProb = random_(0, 4);
+	int num = 0;
 
-	switch (dir == 1 ? dirProb != 0 : dirProb == 0) {
-	case FALSE:
-		num = 0;
+	if ((dir != 1 || dirProb == 0) && (dir == 1 || dirProb != 0)) {

```

Comment:
This looks hard to read. Would this be equivalent?
`(dir == 1 && dirProb == 0) || (dir != 1 && dirProb !=0)`

Reply:
I don't think so but I can setup some test and check some variants.

Reply:
It's the same, applied the change.